### PR TITLE
feat(ONYX-1458): add information and guidance to the saves, follows and alerts screens

### PR DESCRIPTION
**Pull request by dariakoko**
This PR resolves [ONYX-1458] 

### Description

Add information and guidance to the saves, follows and alerts screens. Used InfoButton component where possible, refactored InfoButton component extracting the modal into a separate component

https://github.com/user-attachments/assets/45f4563e-1cf5-45d3-babd-394a5fff7569
 
### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add information and guidance to the saves, follows and alerts screens

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1458]: https://artsyproduct.atlassian.net/browse/ONYX-1458?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 07cabcf5135d92ae0e343b1f7317b372d4a118f8
</p>

File: src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tsx [hunk 1]
```diff
@@ -237,7 +261,48 @@ export const SavedSearchesListWrapper: React.FC<SavedSearchListWrapperProps> = (
...
+            titleElement={
+              <Text variant="lg-display" mr={1}>
+                Alerts
+              </Text>
+            }
+            modalTitle="Alerts"
+            modalContent={
+              <Join separator={<Spacer y={2} />}>
+                <Flex flexDirection="row" alignItems="flex-start">
+                  <TrendingIcon mr={0.5} />
```
Comment (anandaroop) [inline src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tsx:276]: **question:** `TrendingIcon` and `BellIcon` on L285 are reversed compared to [the Figma design](https://www.figma.com/design/zRKGxOWHhq3kcNiNAZCig3/OtH-App%3A-Alerts-and-Activity?node-id=391-12107&m=dev) — is that intentional?

File: src/app/Scenes/Favorites/Favorites.tsx [hunk 2]
```diff
@@ -61,6 +71,48 @@ export const Favorites: React.FC = () => {
...
+                      <Text variant="sm-display">
+                        Tailor your experience, helping you discover artworks that match your taste.
+                      </Text>
+                    </Flex>
+                  </Flex>
+                  <Flex flexDirection="row" alignItems="flex-start">
+                    <BellIcon mr={0.5} />
+                    <Flex flex={1}>
+                      <Text variant="sm-display">
+                        Never miss out by exploring your Activity and receiving timely email updates
```
Comment (anandaroop) [inline src/app/Scenes/Favorites/Favorites.tsx:107]: **question:** Wondering about the consistency of the periods at the end of these sentences. I see that this matches the Figma, but the lack of a period on this one seems inconsistent. (Similar q about the copy in SavesTabHeader)

**Review by anandaroop — DISMISSED** [review]

File: src/app/Scenes/Favorites/Favorites.tsx (hunk 2 above)
Comment (dariakoko) [reply src/app/Scenes/Favorites/Favorites.tsx:107]: I would guess we need them :) Good catch!

**Review by anandaroop — APPROVED** [review]

**Pull request merged**